scripts/update-codes.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * This script should be run after modifying codes.json
+ * It regenerates codes.ts, and updates the table in README.md
+ **/
+
+// eslint-disable no-console
+import fs from 'fs-extra';
+
+import {
+  Writers, VariableDeclarationKind, Project, StructureKind, EnumMemberStructure, OptionalKind,
+} from 'ts-morph';
+
+import markdownTable from 'markdown-table';
+
+import Codes from '../codes.json';
+
+interface jsonCodeComment {
+  doc: string;
+  description: string;
+}
+
+interface jsonCode {
+  code: number;
+  phrase: string;
+  constant: string;
+  comment: jsonCodeComment;
+  isDeprecated?: boolean;
+}
+
+const run = async () => {
+  console.log('Updating src/codes.ts');
+
+  const project = new Project({
+    tsConfigFilePath: 'tsconfig.json',
+  });
+
+  // TODO @prettymuchbryce clean up these horribly unreadable one-liners
```

Comment:
[suggestion] Here's a start, at the least, that you can tweak to your preferences. You could take it a step further and pull out `OptionalKind<EnumMemberStucture>[]` into a type, and even `Record<string, string>` and `Record<string, number`, though neither bother me. I find logic in template literals can become a bit hard to read quickly, so I pulled out any logic, but you don't need to.

```
const reasonPhraseMembers: OptionalKind<EnumMemberStructure>[] = Codes
    .map(({ phrase, constant, comment, isDeprecated }: jsonCode) => {
        const { doc, description } = comment;
        const deprecatedString = isDeprecated ? '@deprecated\n': '';
        return {
            name: constant,
            value: phrase,
            docs: [`${deprecatedString}${doc}\n\n${description}`]
        };
    });

const statusCodeMembers: OptionalKind<EnumMemberStructure>[] = Codes
    .map(({ code, constant, comment, isDeprecated }: jsonCode) => {
        const { doc, description } = comment;
        const deprecatedString = isDeprecated ? '@deprecated\n': '';
        return {
            name: constant,
            value: code,
            docs: [`${deprecatedString}${doc}\n\n${description}`]
        };
    });

const statusCodeToReasonPhrase = Codes
    .reduce((acc: Object, { code, phrase }) => { 
        (acc as Record<string, string>)[`"${code.toString()}"`] = `"${phrase}"`;
        return acc;
    }, {});

const reasonPhraseToStatusCode = Codes
    .reduce((acc: Object, { code, phrase }) => {
        (acc as Record<string, number>)[`"${phrase}"`] = code;
        return acc;
    }, {});
```